### PR TITLE
Cumulative changes from recent PRs

### PR DESCRIPTION
**Pull request by EndymionJkb**
We refactored fee collection, changed some function signatures, and made queries non-payable.

### CODE REVIEW

File: docs/developer-reference/contracts/vault-api.md [hunk 1]
```diff
@@ -1078,9 +1064,9 @@ This `VaultAdmin` function disables recovery mode for a pool. This is a permissi
 ### `quote`
 
 ```solidity
-function quote(bytes calldata data) external payable returns (bytes memory result);
+function quote(bytes calldata data) external returns (bytes memory result);
 ```
-This `VaultExtension` function performs a callback on `msg.sender` with arguments provided in `data`. It is used to query a set of operations on the Vault. Only off-chain `eth_call` are allowed, anything else will revert.
+This `VaultExtension` function performs a callback on `msg.sender` with arguments provided in `data`. It is used to query a set of operations on the Vault. Only off-chain `eth_call` are allowed, anything else will revert. Also not that it is non-payable, as the Vault does not allow ETH.
```
Comment (mkflow27) [inline docs/developer-reference/contracts/vault-api.md]:
```suggestion
This `VaultExtension` function performs a callback on `msg.sender` with arguments provided in `data`. It is used to query a set of operations on the Vault. Only off-chain `eth_call` are allowed, anything else will revert. Also note that it is non-payable, as the Vault does not allow ETH.
```

I think this is supposed to mean note?

File: docs/developer-reference/contracts/vault-api.md [hunk 2]
```diff
@@ -1097,9 +1083,9 @@ This `VaultExtension` function performs a callback on `msg.sender` with argument
 ### `quoteAndRevert`
 
 ```solidity
-function quoteAndRevert(bytes calldata data) external payable;
+function quoteAndRevert(bytes calldata data) external;
 ```
-This `VaultExtension` function performs a callback on `msg.sender` with arguments provided in `data`. It is used to query a set of operations on the Vault. Only off-chain `eth_call` are allowed, anything else will revert. This call always reverts, returning the result in the revert reason.
+This `VaultExtension` function performs a callback on `msg.sender` with arguments provided in `data`. It is used to query a set of operations on the Vault. Only off-chain `eth_call` are allowed, anything else will revert. This call always reverts, returning the result in the revert reason. Also not that it is non-payable, as the Vault does not allow ETH.
```
Comment (mkflow27) [inline docs/developer-reference/contracts/vault-api.md]:
```suggestion
This `VaultExtension` function performs a callback on `msg.sender` with arguments provided in `data`. It is used to query a set of operations on the Vault. Only off-chain `eth_call` are allowed, anything else will revert. This call always reverts, returning the result in the revert reason. Also note that it is non-payable, as the Vault does not allow ETH.
```

I think this is supposed to mean note?

**Review by mkflow27 — APPROVED** [review]
Thanks Jeff for adding these changes.

File: docs/developer-reference/contracts/vault-api.md (hunk 1 above)
Comment (EndymionJkb) [reply docs/developer-reference/contracts/vault-api.md]: Yes, thanks!

**Pull request merged**